Source/BestProjectEver/Private/AI/BPE_PatrolPath.cpp

```diff
@@ -0,0 +1,36 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+
+#include "AI/BPE_PatrolPath.h"
+
+#include "Components/SplineComponent.h"
+
+// Sets default values
+ABPE_PathFollowing::ABPE_PathFollowing()
+{
+	PrimaryActorTick.bCanEverTick = false;
+
+	SplineComponent = CreateDefaultSubobject<USplineComponent>(TEXT("SplineCompononent"));
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+FVector ABPE_PathFollowing::GetWorldLocationByIndex(const int32 Index) const
+{
+	if(IsValid(SplineComponent) && Index >= 0 && Index <= GetLastIndex())
+	{
+		return SplineComponent->GetLocationAtSplinePoint(Index,ESplineCoordinateSpace::World);
+	}
+	return GetActorLocation();
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+int32 ABPE_PathFollowing::GetLastIndex() const
+{
+	if(IsValid(SplineComponent))
+	{
+		return SplineComponent->GetNumberOfSplinePoints();
```

Comment:
This isn't the last index 👀

Reply:
![image](https://user-images.githubusercontent.com/85327912/173107079-4be753e1-9245-4819-b9ce-8681a24e0258.png)
I also corrected this

Reply:
What do you mean? If you corrected the GetLastIndex method, you didn't need to change what you're showing me

Reply:
Because it wasn't the last index so it need to be less than not less or equal

Reply:
Sorry I also I changed this function 
![image](https://user-images.githubusercontent.com/85327912/173114416-a903945d-d160-402b-a3fb-cc91f28531b9.png)